servicetalk-http-utils/src/main/java/io/servicetalk/http/utils/RedirectingHttpRequesterFilter.java

```diff
@@ -162,26 +211,129 @@ public StreamingHttpConnectionFilter create(final FilterableStreamingHttpConnect
             @Override
             public Single<StreamingHttpResponse> request(final HttpExecutionStrategy strategy,
                                                          final StreamingHttpRequest request) {
-                return RedirectingHttpRequesterFilter.this.request(delegate(), strategy, request,
-                        onlyRelativeConnection);
+                return RedirectingHttpRequesterFilter.this.request(delegate(), strategy, request, false);
             }
         };
     }
 
     private Single<StreamingHttpResponse> request(final StreamingHttpRequester delegate,
                                                   final HttpExecutionStrategy strategy,
                                                   final StreamingHttpRequest request,
-                                                  final boolean onlyRelative) {
-        final Single<StreamingHttpResponse> response = delegate.request(strategy, request);
-        if (maxRedirects <= 0) {
+                                                  final boolean allowNonRelativeRedirects) {
+        final Single<StreamingHttpResponse> response = delegate.request(strategy,
+                // Duplicate each payload buffer chunk to allow safely replaying it without worry that indexes can move
```

Comment:
This improvement seems unrelated to the context of this PR – perhaps can be introduced as a separate PR?

Reply:
This PR introduces config options that allow users to carry the payload body forward for each redirect. Without this change, some tests failing when new options are used. So, this is a related change.

Reply:
Thanks for explaining!